File: acct-common-ws-spec/src/main/java/com/desolatetimelines/acct/common/ws/model/AcctPage.java
package com.desolatetimelines.acct.common.ws.model;

import java.util.Collection;

/**
 * Defines a page of elements of a given data type
 *
 * @param data a collection of the elements
 * @param page page information
 * @param <T>  the given data type
 */
public record AcctPage<T>(

    Collection<T> data,

    AcctPageInfo page

) {
}

File: acct-gui-angular/src/app/model-acct/acct-page.ts
import { AcctPageInfo } from "./page-info";

/**
 * Defines a page of elements of a given data type
 */
export interface AcctPage<T> {
    /**
     * a collection of the elements
     */
    data : T[],

    /**
     * page information
     */
    page : AcctPageInfo
}

/**
 * Returns an empty page of elements of the given data type
 */
export function emptyPage<T>() : AcctPage<T> {
    return {
        page: {
            number: 0,
            size: 0,
            totalElements: 0,
            totalPages: 0
          },
          data: []
    }
}